### ISSUE TITLE
Fails to compile w/ GCC 10.x on Ubuntu 18.04

### ISSUE THREAD

**Issue by define-private-public** [copperspice/cs_paint#7]
Hi,

I watched one of your talks about font rendering and I wanted to check this out for one of my projects.  It looks fairly nice.  But when I tried to compile, I received this error:

```
[ 55%] Building CXX object CMakeFiles/CsPaint.dir/src/cspaint_shader.cpp.o
/home/ben/Desktop/cs_paint/src/cspaint_pipeline.cpp: In constructor ‘CsPaint::pipeline::pipeline(CsPaint::device_handle, const surface_handle&, CsPaint::renderpass_handle, const vk::PipelineVertexInputStateCreateInfo&, const std::vector<vk::PipelineShaderStageCreateInfo>&, const vk::PipelineLayout&)’:
/home/ben/Desktop/cs_paint/src/cspaint_pipeline.cpp:95:34: error: no match for ‘operator[]’ (operand types are ‘vk::ResultValue<std::vector<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>, std::allocator<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic> > > >’ and ‘int’)
   95 |    m_pipeline  = std::move(retval[0]);
      |                                  ^
In file included from /usr/include/c++/10/memory:66,
                 from /home/ben/Desktop/cs_paint/src/cspaint_fwd.h:22,
                 from /home/ben/Desktop/cs_paint/src/cspaint_pipeline.h:22,
                 from /home/ben/Desktop/cs_paint/src/cspaint_pipeline.cpp:19:
/usr/include/c++/10/bits/stl_uninitialized.h: In instantiation of ‘_ForwardIterator std::uninitialized_copy(_InputIterator, _InputIterator, _ForwardIterator) [with _InputIterator = __gnu_cxx::__normal_iterator<const vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>*, std::vector<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>, std::allocator<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic> > > >; _ForwardIterator = vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>*]’:
/usr/include/c++/10/bits/stl_uninitialized.h:325:37:   required from ‘_ForwardIterator std::__uninitialized_copy_a(_InputIterator, _InputIterator, _ForwardIterator, std::allocator<_Tp>&) [with _InputIterator = __gnu_cxx::__normal_iterator<const vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>*, std::vector<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>, std::allocator<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic> > > >; _ForwardIterator = vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>*; _Tp = vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>]’
/usr/include/c++/10/bits/stl_vector.h:558:31:   required from ‘std::vector<_Tp, _Alloc>::vector(const std::vector<_Tp, _Alloc>&) [with _Tp = vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>; _Alloc = std::allocator<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic> >]’
/usr/include/vulkan/vulkan.hpp:13434:18:   required from ‘vk::ResultValue<T>::ResultValue(vk::Result, T&) [with T = std::vector<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>, std::allocator<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic> > >]’
/usr/include/vulkan/vulkan.hpp:13579:12:   required from ‘vk::ResultValue<T> vk::createResultValue(vk::Result, T&, const char*, std::initializer_list<vk::Result>) [with T = std::vector<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic>, std::allocator<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic> > >]’
/usr/include/vulkan/vulkan.hpp:76869:29:   required from ‘vk::ResultValue<std::vector<vk::UniqueHandle<vk::Pipeline, Dispatch>, Allocator> > vk::Device::createGraphicsPipelinesUnique(vk::PipelineCache, vk::ArrayProxy<const vk::GraphicsPipelineCreateInfo>, vk::Optional<const vk::AllocationCallbacks>, const Dispatch&) const [with Allocator = std::allocator<vk::UniqueHandle<vk::Pipeline, vk::DispatchLoaderStatic> >; Dispatch = vk::DispatchLoaderStatic]’
/home/ben/Desktop/cs_paint/src/cspaint_pipeline.cpp:94:106:   required from here
/usr/include/c++/10/bits/stl_uninitialized.h:137:72: error: static assertion failed: result type must be constructible from value type of input range
  137 |       static_assert(is_constructible<_ValueType2, decltype(*__first)>::value,
      |                                                                        ^~~~~
CMakeFiles/CsPaint.dir/build.make:302: recipe for target 'CMakeFiles/CsPaint.dir/src/cspaint_pipeline.cpp.o' failed
make[2]: *** [CMakeFiles/CsPaint.dir/src/cspaint_pipeline.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/CsPaint.dir/all' failed
make[1]: *** [CMakeFiles/CsPaint.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

I grabbed the latest version of the source.  the cmake configuration step went fine.

**Comment by agserm** [comment]
Thanks for this report. I have been able to reproduce this on a test machine and am investigating.

Just to confirm, can you list the version of the vulkan headers that are on your machine? You should be able to query this with `dpkg -l libvulkan-dev`.

**Comment by define-private-public** [comment]
`1.2.141.0~rc3-1ubuntu18.04.1.1`

**Comment by agserm** [comment]
Thanks for reporting this, sorry for the delay in updating the ticket. This issue should be resolved in a432b7f.
